plugins/redfish/fu-redfish-legacy-device.c

```diff
@@ -72,6 +107,214 @@ fu_redfish_legacy_device_attach(FuDevice *dev, FuProgress *progress, GError **er
+	g_autoptr(FuRedfishRequest) request = fu_redfish_backend_request_new(backend);
+
+	/* create URI and poll */
+	if (!fu_redfish_request_perform(request,
+					"/redfish/v1/UpdateService",
+					FU_REDFISH_REQUEST_PERFORM_FLAG_LOAD_JSON,
+					error))
+		// If we cannot poll the BMC, let's try again, it's probably updating itself
+		return TRUE;
+
+	/* percentage is optional */
+	json_obj = fu_redfish_request_get_json_object(request);
+
+	if (json_object_has_member(json_obj, "Oem")) {
+		JsonObject *oem = json_object_get_object_member(json_obj, "Oem");
+		if (oem != NULL && json_object_has_member(oem, "Hpe")) {
+			JsonObject *hpe = json_object_get_object_member(oem, "Hpe");
+
+			const gchar *status = json_object_get_string_member(hpe, "State");
+			if (g_strcmp0(status, "Error") == 0) {
+				if (json_object_has_member(hpe, "Result")) {
+					JsonObject *result =
+					    json_object_get_object_member(hpe, "Result");
+					const gchar *message_id = NULL;
+
+					if (json_object_has_member(result, "MessageId"))
+						message_id =
+						    json_object_get_string_member(result,
+										  "MessageId");
+					if (json_object_has_member(result, "Message"))
+						message = json_object_get_string_member(result,
+											"Message");
+					/* use the message */
+					g_debug("message [%s]: %s", message_id, message);
+					fu_redfish_device_poll_set_message_id(self,
+									      ctx,
+									      message_id);
+					return FALSE;
+				}
+			}
+
+			gint64 pc = json_object_get_int_member(hpe, "FlashProgressPercent");
+			if (pc >= 0 && pc <= 100)
+				fu_progress_set_percentage(ctx->progress, (guint)pc);
+
+			if (g_strcmp0(status, "Writing") == 0 ||
+			    g_strcmp0(status, "Updating") == 0) {
+				fu_progress_set_status(ctx->progress, FWUPD_STATUS_DEVICE_WRITE);
+			} else if (g_strcmp0(status, "Verifying") == 0) {
+				fu_progress_set_status(ctx->progress, FWUPD_STATUS_DEVICE_VERIFY);
+			} else if (g_strcmp0(status, "Complete") == 0) {
+				ctx->completed = TRUE;
+				fu_progress_set_status(ctx->progress, FWUPD_STATUS_IDLE);
+			}
+		}
+	}
+
+	/* try again */
+	return TRUE;
+}
+
+static gboolean
+fu_redfish_legacy_device_hpe_write_firmware(FuDevice *device,
+					    GBytes *fw,
+					    FuProgress *progress,
+					    FwupdInstallFlags flags,
+					    GError **error)
+{
+	FuRedfishLegacyDevice *self = FU_REDFISH_LEGACY_DEVICE(device);
+	FuRedfishBackend *backend = fu_redfish_device_get_backend(FU_REDFISH_DEVICE(self));
+	CURL *curl;
+	JsonObject *json_obj;
+	curl_mimepart *part;
+	gchar *sessionkey;
+	g_autoptr(curl_mime) mime = NULL;
+	g_autoptr(FuRedfishRequest) request = NULL;
+	g_autoptr(_curl_slist) hs = NULL;
+
+	g_autoptr(GString) json_str = g_string_new(NULL);
+	g_autoptr(JsonBuilder) builder = json_builder_new();
+	g_autoptr(JsonGenerator) json_generator = json_generator_new();
+	g_autoptr(JsonNode) json_root = NULL;
+
+	if (!fu_redfish_legacy_device_hpe_is_ready(FU_REDFISH_DEVICE(self), NULL, error)) {
+		return FALSE;
+	}
```

Comment:
no need for braces

Reply:
For my curiosity, what's the rule of "when" and "when not"?
I'm referencing [this](https://github.com/fwupd/fwupd/pull/8478#discussion_r1962308935) comment from Richard.

Reply:
I think only braces when it's ambiguous what's inside the conditional.